test/unit-tests/path.mlt

```diff
@@ -312,3 +312,14 @@ Path.(reach_for_running (relative root "foo") ~from:(Path.relative root "foo"))
 [%%expect{|
 - : string = "./."
 |}]
+
+Path.relative Path.root "_build"
+[%%expect{|
+- : Stdune.Path.t = (In_build_dir ".")
+|}]
+
+(* This is not right, but kind of annoying to fix :/ *)
+Path.relative (r "foo") "../_build"
```

Comment:
Previously, I tried addressing this by having logic in the `in_source_tree` smart constructor to automatically change such paths to `In_build_dir`. This wasn't really necessary to make the feature work, so I dropped it. In any case, I wasn't comfortable with that approach either.